realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -637,36 +637,49 @@ public void setList(String fieldName, RealmList<DynamicRealmObject> list) {
             throw new IllegalArgumentException("Null values not allowed for lists");
         }
 
-        String tableName = proxyState.getRow$realm().getTable().getName();
+        long columnIndex = proxyState.getRow$realm().getColumnIndex(fieldName);
```

Comment:
The answer is yes since `Row` object in `DynamicRealmObject` is always `CheckedRow` or `InvalidRow`.

Despite being written in the Javadoc of `Row` class (`return the index, -1 if not found`), `CheckedRow` always checks the existence of the row and throws `IllegalArgumentException` if the column does not exist.